src/main/java/io/fabric8/jenkins/openshiftsync/BuildConfigToJobMapper.java

```diff
@@ -165,23 +172,27 @@ public static boolean updateBuildConfigFromJob(WorkflowJob job, BuildConfig buil
                     ref = branch;
                   }
                 }
+                LOGGER.info("updating bc " + namespaceName + " gitURL to " + gitUrl + " and ref to " + ref);
                 OpenShiftUtils.updateGitSourceUrl(buildConfig, gitUrl, ref);
+                rc = true;
               }
             }
           }
         }
-        return true;
+        return rc;
       }
       return false;
     }
 
     if (definition instanceof CpsFlowDefinition) {
       CpsFlowDefinition cpsFlowDefinition = (CpsFlowDefinition) definition;
       String jenkinsfile = cpsFlowDefinition.getScript();
-      if (jenkinsfile != null && jenkinsfile.trim().length() > 0) {
+      if (jenkinsfile != null && jenkinsfile.trim().length() > 0 && !jenkinsfile.equals(jenkinsPipelineStrategy.getJenkinsfile())) {
+        LOGGER.info("updating bc " + namespaceName + " jenkinsfile to " + jenkinsfile + " where old jenkinsfile was " + jenkinsPipelineStrategy.getJenkinsfile());
```

Comment:
i originally did have them lower, then saw use of logger.info in a bunch of other places ... decided to be consistent vs. undertake a logging overhaul with this pull